src/hotspot/share/runtime/abstract_vm_version.cpp

```diff
@@ -138,31 +138,39 @@ const char* Abstract_VM_Version::vm_vendor() {
 
 
 const char* Abstract_VM_Version::vm_info_string() {
+  const char* mode;
   switch (Arguments::mode()) {
     case Arguments::_int:
-      return CDSConfig::is_using_archive() ? "interpreted mode, sharing" : "interpreted mode";
+      mode = "interpreted mode";
+      break;
     case Arguments::_mixed:
-      if (CDSConfig::is_using_archive()) {
-        if (CompilationModeFlag::quick_only()) {
-          return "mixed mode, emulated-client, sharing";
-        } else {
-          return "mixed mode, sharing";
-         }
+      if (CompilationModeFlag::quick_only()) {
+        mode = "mixed mode, emulated-client";
       } else {
-        if (CompilationModeFlag::quick_only()) {
-          return "mixed mode, emulated-client";
-        } else {
-          return "mixed mode";
-        }
+        mode = "mixed mode";
       }
+      break;
     case Arguments::_comp:
       if (CompilationModeFlag::quick_only()) {
-         return CDSConfig::is_using_archive() ? "compiled mode, emulated-client, sharing" : "compiled mode, emulated-client";
+        mode = "compiled mode, emulated-client";
+      } else {
+        mode = "compiled mode";
       }
-      return CDSConfig::is_using_archive() ? "compiled mode, sharing" : "compiled mode";
+      break;
+    default:
+      ShouldNotReachHere();
   }
-  ShouldNotReachHere();
-  return "";
+
+  const char* static_info = ", static";
+  const char* sharing_info = ", sharing";
+  size_t len = strlen(mode) + (is_vm_statically_linked() ? strlen(static_info) : 0) +
+               (CDSConfig::is_using_archive() ? strlen(sharing_info) : 0) + 1;
```

Comment:
Make it easier to add new cases:

```suggestion
  size_t len = strlen(mode) +
               (is_vm_statically_linked() ? strlen(static_info) : 0) +
               (CDSConfig::is_using_archive() ? strlen(sharing_info) : 0) + 
               1;
```

Reply:
Done.